### ISSUE TITLE
No option for default section category

### ISSUE THREAD

**Issue by bradcdelaney** [alleyinteractive/apple-news#322]
Hey, thank you so much for the category mapping! That was a big help. There's only one issue... there doesn't seem to be a way to map all categories to a default section - like there is on the back end of Apple News... at least from what I can see! So basically we have subsections but like having a "Latest" section with all of our articles. That doesn't seem to be happening since the sections mapping.

Thanks!

-Brad

**Comment by bcampeau** [comment]
The plugin automatically selects your default Apple News section if no mappings exist, which is usually called "Main". It sounds like what you're asking is for the ability to set a different default section other than Main.

Is this correct?

**Comment by bradcdelaney** [comment]
No,

Here are the sections we have set up: latest, politics, entertainment, lifestyle, sports. 

We like all of our posts to list under the latest section as well as under their respective category section. So a default section would be nice so all of our posts automatically tag to latest and then we can select additional categories that they might fall into.

We had it set this way prior to mapping our categories but now anything that has a mapped category is only being distributed to the mapped section and not hitting our 'latest' section at all.

**Comment by bcampeau** [comment]
We'll consider this for a future plugin version. In the meantime, feel free to take advantage of one of the [available filters](https://github.com/alleyinteractive/apple-news/wiki/Action-and-Filter-Hooks#includesapple-push-apirequestclass-requestphp) to customize the sections however you'd like. `apple_news_post_args` is probably the one you're looking for.

**Comment by kujain** [comment]
In the same boat here as well! The Main section only shows those content that are not assigned to specific sections (ie leftovers). Rather than using the UI to manually add all categories to Main, is there a filter/hook to automate the section/category assignment for main? Could you provide an example for using `apple_news_post_args` above? Thanks!

**Comment by kevinfodness** [comment]
@kujain - the plugin should be assigning any posts without a category mapping explicitly set to the Main section automatically, plus anything you've manually mapped to Main. Is this not happening? Can you provide steps to reproduce? The request above was to change the default category to something else via the UI, but it sounds like you are interested in doing something different.

**Comment by kujain** [comment]
Thanks for the reply @kevinfodness !
The issue I am having (and I don't believe its a bug - rather its by design) is this:

I have mapped my WP categories to defined sections. **Main** has no category assigned, but there's nothing specifically assigned to Main: 
![image](https://user-images.githubusercontent.com/120631/70149815-e01a6680-1676-11ea-816d-1eafc1dbc1aa.png)

But the AppleNews channel page uses **Main** as its main landing page. Since we have nothing assigned to main, only the content not explicitly assigned to main (through default mapping) is being pushed to the Channel landing page, while we would have preferred ALL content to be pushed to **Main**, regardless of the default Section mapping. Hence apple news readers accessing our channel page are only seeing content that do not belong to the specific sections (ie outliers), rather than seeing this page as a collection of all our stories!

Some ways we could workaround this:
- Create a new Category in WP called Main or Featured and then assign ALL posts to this category during editing/publishing. Then map this category to **Main**. Now all content that's explicitly assigned Main or Featured category will show up here, along with on any other mapped section. Of course, this requires additional workflow and considerations and not ideal.
- Use the Apple News Sidebar to manually assign to specific sections, including Main. Of course, this rather defeats the purpose of the default mapping as we would need to do this for all content!
- Map ALL of our WP categories to the Main section along with other section mappings. This should ideally work, but considering we have about 25 categories (including subcategories) it is not the best UI and there are issues with this since the section mapping doesn't bind by category ID, rather just by its name (as [referred here](https://github.com/alleyinteractive/apple-news/issues/673)).
- Use a hook to manually auto-assign a content to Main section during publishing. This was what I was hoping to find as it will help resolve this issue. I looked into `apple_news_generate_json` but i didnt counldn't identify where it assigns the content to sections.

Thanks very much!

**Comment by kevinfodness** [comment]
OK, yes, that makes sense. Did you look into using the `apple_news_post_args` filter to modify the section mapping?

**Comment by kujain** [comment]
Thanks @kevinfodness !
Could you provide me an example snippet for using `apple_news_post_args` ? I see it was mentioned earlier in this thread but I was not sure how to apply this filter and use it for my purpose...

**Comment by kujain** [comment]
hi @kevinfodness I was able to make some progress. I believe that I need to use `apple_news_api_post_meta` instead since it gives me access to the metadata for the article which contains the links to the sections:
```
(
    [data] => Array
        (
            [links] => Array
                (
                    [sections] => Array
                        (
                            [0] => https://news-api.apple.com/sections/e29wew34d4896-17cf-489d-ba95-047f1f7f8dc2
                        )

                )

            [isPaid] => 
            [isPreview] => 
            [isHidden] => 
            [isSponsored] => 
        )

)
```
Assuming that's correct, I would need to use the filter to insert the Main section url into the sections array. But how can I find out the Main section url from within my filter/custom function? Is there a way for me to access the \Apple_Actions\Index\Section->get_sections() method from my theme files, or is there any way to get this info? 
Thanks!

**Comment by kujain** [comment]
I was able to access the Apple_News functions and update the Sections meta. Here's the code I used for reference/feedback. 
```
		if ( isset($meta['data']['links']['sections']) ) {

			// connect to Apple News plugin methods
			$admin_settings = new \Admin_Apple_Settings();
			$section_api    = new \Apple_Actions\Index\Section( $admin_settings->fetch_settings() );
			$sections_raw   = $section_api->get_sections();

			// find the url for the Main section
			foreach ( $sections_raw as $section ) {
				if ( ! empty( $section->id ) && ! empty( $section->name ) && true === $section->isDefault ) {
					$section_default_url = $section->links->self;
					break;
				}
			}

			// insert if not already present in metadata
			if ( isset( $section_default_url ) && true !== in_array( $section_default_url, $meta['data']['links']['sections'] ) ) {
				$meta['data']['links']['sections'][] = $section_default_url;
			}
		}

		return $meta;
	}
```
Now all pushed content are added to Main as well as any other default sections! Thanks!

**Comment by kevinfodness** [comment]
We just released version 2.1.0 which includes some fixes to section mapping. If you have a section that is mapped to another category and you also want it to appear on Main, then assign it to Main also, and it will publish to both.
